### PR TITLE
Delete serverless directory on backport PRs

### PR DESCRIPTION
**Pull request by bmorelli25**
Simplified version of https://github.com/elastic/observability-docs/pull/3889 that ~doesn't work for some reason. Introduces conflicts.~ works after @colleenmcginnis fixed it.

### CODE REVIEW

File: .github/workflows/delete-serverless-dir-on-backports.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Delete serverless directory in backports
+
+on:
+  pull_request:
+    branches:
+      - '7.*'
+      - '8.*'
+    branches-ignore:
+      - 'main'
```
Comment (bmorelli25) [inline .github/workflows/delete-serverless-dir-on-backports.yml:9]:
```suggestion
```

**Review by colleenmcginnis — COMMENTED** [review]
This looks good to me. Can you add a note to [the README](https://github.com/elastic/observability-docs?tab=readme-ov-file#backporting) about this? It would be good to have a place to point contributors if they notice this behavior and have questions about this.

**Comment by bmorelli25** [comment]
Per Joe's testing, this needs to be backported to any branch where it will be used.

**Review by colleenmcginnis — APPROVED** [review]
🐦

**Pull request merged**